File: JDApp-Java/other/src/main/java/com/jd/other/application/OtherApplication.java
package com.jd.other.application;

import android.app.Application;
import android.util.Log;

import com.renny.mylibrary.IAppInit;

/*
    @see https://github.com/ren93/initiator
 */

public class OtherApplication implements IAppInit {
    @Override
    public void init(Application application) {
        Log.d("init==", "OtherApplication");
    }
}

File: JDApp-Kotlin/other/src/main/java/com/jd/other/application/OtherApplication.kt
package com.jd.other.application

import android.app.Application
import android.content.Context
import android.util.Log
import com.google.auto.service.AutoService
import com.jd.core.base.IApplication


@AutoService(IApplication::class)
class OtherApplication : IApplication {
    override fun onAttachBaseContext(context: Context) {

    }

    override fun onCreate(application: Application) {
        Log.d("init==", "OtherApplication")
    }

    override fun onTerminate(application: Application) {

    }

    override fun onAsyncInit() {

    }
}
